plume-front/src/editor.rs

```diff
@@ -62,7 +69,183 @@ impl From<stdweb::private::ConversionError> for EditorError {
+        console!(log, "Got subtitle");
+        subtitle.raw_value()
+    } else {
+        let subtitle_element = HtmlElement::try_from(
+            document()
+                .query_selector("#plume-editor > h2")
+                .unwrap()
+                .unwrap(),
+        )
+        .ok()
+        .unwrap();
+        subtitle_element.inner_text()
+    }
+}
+fn get_tags() -> String {
+    let tags: InputElement = InputElement::try_from(document().get_element_by_id("tags").unwrap())
+        .ok()
+        .unwrap();
+    tags.raw_value()
+}
+fn get_license() -> String {
+    let license: InputElement =
+        InputElement::try_from(document().get_element_by_id("license").unwrap())
+            .ok()
+            .unwrap();
+    console!(log, "Got license");
+    license.raw_value()
+}
+fn get_cover() -> String {
+    let cover: SelectElement =
+        SelectElement::try_from(document().get_element_by_id("cover").unwrap())
+            .ok()
+            .unwrap();
+    console!(log, "Got cover");
+    cover.raw_value()
+}
+fn autosave() {
+    let info: AutosaveInformation = AutosaveInformation {
+        contents: get_editor_contents(),
+        title: get_title(),
+        subtitle: get_subtitle(),
+        tags: get_tags(),
+        license: get_license(),
+        last_saved: Date::now(),
+        cover: get_cover(),
+    };
+    let id = get_autosave_id();
+    match window()
+        .local_storage()
+        .insert(&id, &serde_json::to_string(&info).unwrap())
+    {
+        Ok(_) => {}
+        _ => console!(log, "Autosave failed D:"),
+    }
+}
+//This is only necessary until we go to stdweb 4.20 at least
+fn confirm(message: &str) -> bool {
+    return js! {return confirm(@{message});} == true;
+}
+fn load_autosave() {
+    if let Some(autosave_str) = window().local_storage().get(&get_autosave_id()) {
+        let autosave_info: AutosaveInformation = serde_json::from_str(&autosave_str).ok().unwrap();
+        let message = format!(
+            "{} {}?",
+            i18n!(
+                CATALOG,
+                "Do you want to load the local autosave last edited at"
+            ),
+            Date::from_time(autosave_info.last_saved).to_date_string()
+        );
+        if confirm(&message) {
+            set_value("editor-content", &autosave_info.contents);
+            set_value("title", &autosave_info.title);
+            set_value("subtitle", &autosave_info.subtitle);
+            set_value("tags", &autosave_info.tags);
+            set_value("license", &autosave_info.license);
+            set_value("cover", &autosave_info.cover);
+            console!(log, "Loaded autosave.");
+        } else {
+            clear_autosave();
+        }
+    }
+}
+fn clear_autosave() {
+    window().local_storage().remove(&get_autosave_id());
+    console!(log, &format!("Saved to {}", &get_autosave_id()));
+}
+static mut AUTOSAVE_TIMEOUT: Option<TimeoutHandle> = None;
 
+unsafe fn autosave_debounce(interval: u32) {
```

Comment:
Why does it need to be `unsafe`?

Reply:
Oh, it should probably just be unsafe on the lines that refer to the `static mut`

Reply:
No, I don't know much about unsafe, so maybe you are right and whole function should be. But also, I feel like there is a better way to do this, like going with lazy_static for instance.

Reply:
That option looked off to us too tbh.. At least on its own
Maybe it'd be better to use a RefCell or something?

Reply:
Anything that will prevent having `unsafe` code is fine (not that it is really unsafe, we only have one thread when running in the browser).